src/View/Form/EntityContext.php

```diff
@@ -426,6 +426,34 @@ public function isRequired($field)
         return false;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getErrorMessage($field)
+    {
+        $parts = explode('.', $field);
+
+        $validator = $this->_getValidator($parts);
+        $fieldName = array_pop($parts);
+        if (!$validator->hasField($fieldName)) {
```

Comment:
Can the code ever fail to find a validator, resulting in a fatal error here?

Reply:
Not that I can see. It always returns a validator. This logic was copied from `isRequired` which would suffer from the same error.

Reply:
You could however add a sanity check to be safe. And throw meaningful exception (with clear text) then instead of running into a fatal null->usage error.

Reply:
Maybe https://api.cakephp.org/3.5/source-class-Cake.View.Form.EntityContext.html#442-477 could be modified instead to throw an exception (if BC is required, by setting a 2nd parameter to true, which should in 4.x then be the default, in 3.x it would not for existing calls)?